Update bpython to 0.17

flokli opened the pull request:
###### Motivation for this change
As mentioned by @adisbladis, `pythonPackages` is a mess and needs to be cleaned up.

I bumped the `bpython` package to 0.17, cleaned it up, and added `curtsies` and `pyte`, which are both dependencies of `bpython` (at least now)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

flokli committed:
pyte: init at 0.7.0

flokli committed:
curtsies: init at 0.2.11

flokli committed:
bpython: 0.12 -> 0.17

FRidh commented on pkgs/development/python-modules/pyte/default.nix:
```diff
@@ -0,0 +1,29 @@
+
+  propagatedBuildInputs = [ wcwidth ];
+
+  buildInputs = [ pytest pytestrunner ];
```
checkInputs

flokli replied:
done

FRidh commented on pkgs/development/python-modules/bpython/default.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "13fyyx06645ikvmj9zmkixr12kzk1c3a3f9s9i2rvaczjycn82lz";
+  };
+
+  buildInputs = [ mock ];
```
checkInputs

flokli replied:
done

FRidh commented on pkgs/development/python-modules/bpython/default.nix:
```diff
@@ -0,0 +1,28 @@
+  name  = "${pname}-${version}";
+
+  # 0.17 is still missing on PyPI, https://github.com/bpython/bpython/issues/706
+  src = fetchurl {
```
fetchPypi

flokli replied:
0.17 is not released to PyPI yet, thats why `fetchUrl` is used. Issue is linked above.

FRidh commented on pkgs/development/python-modules/curtsies/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+  propagatedBuildInputs = [ blessings wcwidth pyte ];
+
+  buildInputs = [ nose mock pytest ];
```
checkInputs

flokli replied:
done

FRidh commented on pkgs/development/python-modules/bpython/default.nix:
```diff
@@ -0,0 +1,28 @@
+    homepage = "https://bpython-interpreter.org/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ flokli ];
+    platforms = platforms.all;
```
platforms needs to be removed.

flokli replied:
done

FRidh commented on pkgs/development/python-modules/curtsies/default.nix:
```diff
@@ -0,0 +1,28 @@
+    homepage = https://pypi.python.org/pypi/curtsies;
+    license = licenses.mit;
+    maintainers = with maintainers; [ flokli ];
+    platforms = platforms.all;
```
platforms needs to be removed.

flokli replied:
done

FRidh commented on pkgs/development/python-modules/pyte/default.nix:
```diff
@@ -0,0 +1,29 @@
+    homepage = https://github.com/selectel/pyte;
+    license = licenses.lgpl3;
+    maintainers = with maintainers; [ flokli ];
+    platforms = platforms.all;
```
platforms needs to be removed.

flokli replied:
done

FRidh reviewed (commented):
platforms is set by `buildPythonPackage`.

flokli committed:
urwid: move into separate file, use fetchPypi instead of fetchurl

flokli commented:
@FRidh Thanks for your review. I addressed all the problems mentioned.
`bpython-gtk` was removed since `0.14`, and `urwid` is already there, too.
So I added it as `propagatedBuildInput`, and it also runs.

FRidh commented:
@GrahamcOfBorg python3.pkgs.bpython
